python/tank/platform/bundle.py

```diff
@@ -264,6 +283,14 @@ def frameworks(self):
     ##########################################################################################
     # public methods
 
+    def change_context(self, new_context):
+        """
+        Abstract method for context changing. If this has not been implemented
+        in the concrete class requesting the context change then NotImplementedError
+        is raised.
```

Comment:
Maybe docstring should add "Needs to be implemented for all apps that return True from `context_change_allowed`?"